y/zstd.go

```diff
@@ -19,18 +17,38 @@
 package y
 
 import (
-	"github.com/DataDog/zstd"
+	"sync"
+
+	"github.com/klauspost/compress/zstd"
 )
 
-// CgoEnabled is used to check if CGO is enabled while building badger.
-const CgoEnabled = true
+var (
+	zstdDec *zstd.Decoder
+	zstdEnc *zstd.Encoder
+
+	zstdEncOnce, zstdDecOnce sync.Once
+)
 
 // ZSTDDecompress decompresses a block using ZSTD algorithm.
 func ZSTDDecompress(dst, src []byte) ([]byte, error) {
-	return zstd.Decompress(dst, src)
+	var err error
+	zstdDecOnce.Do(func() {
+		zstdDec, err = zstd.NewReader(nil)
+		AssertTrue(err == nil)
+	})
+	return zstdDec.DecodeAll(src, dst)
```

Comment:
```suggestion
	return zstdDec.DecodeAll(src, dst[:0])
```
If you don't want it appended.